packages/core/src/domain/configuration/transportConfiguration.ts

```diff
@@ -1,48 +1,56 @@
+  rumEndpointBuilder: EndpointBuilder
+  sessionReplayEndpointBuilder: EndpointBuilder
+  internalMonitoringEndpointBuilder?: EndpointBuilder
+  isIntakeEndpoint: (url: string) => boolean
   // only on staging build mode
   replica?: ReplicaConfiguration
 }
 
 export interface ReplicaConfiguration {
   applicationId?: string
-  logsEndpoint: string
-  rumEndpoint: string
-  internalMonitoringEndpoint: string
+  logsEndpointBuilder: EndpointBuilder
+  rumEndpointBuilder: EndpointBuilder
+  internalMonitoringEndpointBuilder: EndpointBuilder
 }
 
 export function computeTransportConfiguration(
   initConfiguration: InitConfiguration,
   buildEnv: BuildEnv
 ): TransportConfiguration {
-  const endpointBuilder = createEndpointBuilder(initConfiguration, buildEnv)
-  const intakeUrls: string[] = ENDPOINTS_TYPES.map((endpointType) => endpointBuilder.buildIntakeUrl(endpointType))
+  const endpointBuilders = {
+    logsEndpointBuilder: createEndpointBuilder(initConfiguration, buildEnv, 'logs'),
+    rumEndpointBuilder: createEndpointBuilder(initConfiguration, buildEnv, 'rum'),
+    sessionReplayEndpointBuilder: createEndpointBuilder(initConfiguration, buildEnv, 'sessionReplay'),
+  }
+  const intakeEndpoints: string[] = objectValues(endpointBuilders).map((builder) => builder.buildIntakeUrl())
 
   const configuration: TransportConfiguration = {
-    isIntakeUrl: (url: string) => intakeUrls.some((intakeUrl) => url.indexOf(intakeUrl) === 0),
-    logsEndpoint: endpointBuilder.build('logs'),
-    rumEndpoint: endpointBuilder.build('rum'),
-    sessionReplayEndpoint: endpointBuilder.build('sessionReplay'),
+    isIntakeEndpoint: (url) => intakeEndpoints.some((intakeEndpoint) => url.indexOf(intakeEndpoint) === 0),
+    ...endpointBuilders,
   }
 
   if (initConfiguration.internalMonitoringApiKey) {
-    configuration.internalMonitoringEndpoint = endpointBuilder.build('logs', 'browser-agent-internal-monitoring')
+    configuration.internalMonitoringEndpointBuilder = createEndpointBuilder(
+      initConfiguration,
+      buildEnv,
+      'logs',
+      'browser-agent-internal-monitoring'
+    )
   }
 
   if (buildEnv.buildMode === BuildMode.E2E_TEST) {
-    configuration.internalMonitoringEndpoint = '<<< E2E INTERNAL MONITORING ENDPOINT >>>'
-    configuration.logsEndpoint = '<<< E2E LOGS ENDPOINT >>>'
-    configuration.rumEndpoint = '<<< E2E RUM ENDPOINT >>>'
-    configuration.sessionReplayEndpoint = '<<< E2E SESSION REPLAY ENDPOINT >>>'
+    const e2eEndpointBuilder = (fake: string) => ({ build: () => fake } as EndpointBuilder)
+    configuration.internalMonitoringEndpointBuilder = e2eEndpointBuilder('<<< E2E INTERNAL MONITORING ENDPOINT >>>')
+    configuration.logsEndpointBuilder = e2eEndpointBuilder('<<< E2E LOGS ENDPOINT >>>')
+    configuration.rumEndpointBuilder = e2eEndpointBuilder('<<< E2E RUM ENDPOINT >>>')
+    configuration.sessionReplayEndpointBuilder = e2eEndpointBuilder('<<< E2E SESSION REPLAY ENDPOINT >>>')
```

Comment:
FMU with this new logic we won't send dd-request-id and batch time (as well as all other query parameters) to the e2e intakes. It seems good for now but may be worth revisiting in the future.

Reply:
yes, sure